internal/cluster/cluster.go

```diff
@@ -952,9 +952,54 @@ func (m *Manager) SetVipsData(ctx context.Context, c *common.Cluster, apiVip, in
 	return nil
 }
 
+func (m *Manager) uploadClusterDataAsFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	log := logutil.FromContext(ctx, m.log)
+	events, err := m.eventsHandler.V2GetEvents(ctx, c.ID, nil, nil)
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+
+	eventsAsData, err := json.MarshalIndent(events, "", " ")
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+	fileName := fmt.Sprintf("%s/logs/cluster/events.json", c.ID)
+
+	err = objectHandler.Upload(ctx, eventsAsData, fileName)
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload events file")
+		return err
+	}
+
+	cluster := *c
+	cluster.PullSecret = "SECRET"
+	clusterData, err := json.MarshalIndent(cluster, "", " ")
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload cluster metadata file")
+		return err
+	}
+	fileName = fmt.Sprintf("%s/logs/cluster/metadata.json", c.ID)
+
+	err = objectHandler.Upload(ctx, clusterData, fileName)
+	if err != nil {
+		log.WithError(err).Warn("Failed to upload cluster metadata file")
+		return err
+	}
+
+	return nil
+}
+
 func (m *Manager) CreateTarredClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) (string, error) {
 	log := logutil.FromContext(ctx, m.log)
 	fileName := fmt.Sprintf("%s/logs/cluster_logs.tar", c.ID)
+
+	err := m.uploadClusterDataAsFiles(ctx, c, objectHandler)
```

Comment:
Unsure this belongs here (perhaps it should go to `prepareClusterLogs`), this logic makes the function name a bit misleading. 
Consider logging an error in case the `uploadClusterDataAsFiles` fails instead of immediately returning an error. so an error here (it fail to get the cluster events, etc ) will not prevent the collection of other logs